src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -318,6 +321,209 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
+    {
+        var text = expression;
+        var currentDateTimeOffset = DateTimeOffset.UtcNow;
+
+        // We just pre-matched the prefix, so we need to re-match the full pattern to get the type, offset, and option
+        matches = pattern.Matches(text);
+
+        string format;
+        DateTimeOffset dateTimeOffset = currentDateTimeOffset;
+        var match = matches.FirstOrDefault();
+        if (match?.Groups.Count == 4)
+        {
+            try
+            {
+                IFormatProvider formatProvider = Thread.CurrentThread.CurrentUICulture;
+
+                Group type = match.Groups["type"];
+                if (match.Groups["offset"].Success && match.Groups["option"].Success)
+                {
+                    if (int.TryParse(match.Groups["offset"].Value, out int offset) && offset != 0)
+                    {
+                        dateTimeOffset = dateTimeOffset.AddOffset(offset, match.Groups["option"].Value);
+                    }
+                }
+
+                if (string.Equals(type.Value, "rfc1123", StringComparison.OrdinalIgnoreCase))
+                {
+                    // For RFC1123, we want to be sure to use the invariant culture,
+                    // since we are potentially overriding the format for local date time
+                    // we should explicitly set the format provider to invariant culture
+                    formatProvider = CultureInfo.InvariantCulture;
+                    format = "r";
+                }
+                else if (string.Equals(type.Value, "iso8601", StringComparison.OrdinalIgnoreCase))
+                {
+                    format = "o";
+                }
+                else
+                {
+                    format = type.Value.Substring(1, type.Value.Length - 2);
+                }
+
+                text = dateTimeOffset.ToString(format, formatProvider);
+
+                if(!DateTimeOffset.TryParse(text, out dateTimeOffset) )
+                {
+                    return node.CreateBindingFailure(HttpDiagnostics.DateTimePatternMatchError(expression));
+                }
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                return node.CreateBindingFailure(HttpDiagnostics.DateTimePatternMatchError(expression));
+            }
+            return node.CreateBindingSuccess(text);
+        } 
+        else
+        {
+            return node.CreateBindingFailure(HttpDiagnostics.DateTimePatternMatchError(expression));
+        } 
+
+        
+    }
+
+    private HttpBindingResult<object?> GetRandInt(HttpExpressionNode node, string text, MatchCollection matches)
+    {
+
+        Random random = new();
+        
+        var match = matches.FirstOrDefault();
+        if(match != null)
+        {
+            (int? Min, int? Max)? parameters = ParseParametersFromMatch(match);
+
+            if (parameters is not null)
```

Comment:
nit: I am probably being pedantic, but I think you may want to use the term `argument(s)` instead of `parameter(s)` here for the variable name. See https://stackoverflow.com/questions/156767/whats-the-difference-between-an-argument-and-a-parameter for some discussion around the distinction.